### ISSUE TITLE
Non-Full-Trust environments throw System.Security.VerificationException on simple formula calculation

### ISSUE THREAD

**Issue by jochenwezel** [EPPlusSoftware/EPPlus#314]
## Problem description

* Given is a cell with a formula "500*4" and a call to calculate this cell
* On some framework versions and different trust-environments, the .NET runtime throws a `System.Security.VerificationException : Operation could destabilize the runtime` on entering method `public override CompileResult Compile()`
  * System.Security.VerificationException is catched internally and the resulting cell error value will be `#VALUE!` (instead of `2000`)
* In fact, the compiler shows an 'information' item `Use coalesce expression (IDE0030)` pointing to that same method.
* Bug is related to all EPPlus 4.5 as well as EPPlus 5.x

### Code before patch
```
        public override CompileResult Compile()
        {
            double result = _compiledValue.HasValue ? _compiledValue.Value : double.Parse(ExpressionString, CultureInfo.InvariantCulture);
            result = _negate ? result*-1 : result;
            return new CompileResult(result, DataType.Integer);
        }
```

### Code after patch
```
        public override CompileResult Compile()
        {
            double result = _compiledValue ?? double.Parse(ExpressionString, CultureInfo.InvariantCulture);
            result = _negate ? result*-1 : result;
            return new CompileResult(result, DataType.Integer);
        }
```

### Code before patch
```
        public override CompileResult Compile()
        {
            double result = _compiledValue.HasValue ? _compiledValue.Value : double.Parse(ExpressionString, CultureInfo.InvariantCulture);
            result = _negate ? result * -1 : result;
            return new CompileResult(result, DataType.Decimal);
        }
```

### Code after patch
```
        public override CompileResult Compile()
        {
            double result =  _compiledValue ?? double.Parse(ExpressionString, CultureInfo.InvariantCulture);
            result = _negate ? result * -1 : result;
            return new CompileResult(result, DataType.Decimal);
        }
```

**Comment by swmal** [comment]
Thank you, I’ll have a look at this.

**Comment by swmal** [comment]
I have changed this in the two Expressions and it will be included in v 5.6. We will have to go through the entire codebase, because we have more of these... Will happen, but not sure if it will be done until 5.6.

**Comment by jochenwezel** [comment]
It would be great to see this issue fixed, as it's the main blocker reason for us before further testing can continue for upgrading from v4.5 to an up-to-date release  :-)

**Comment by swmal** [comment]
Hi,
Apologies for the late reply to your last messages. @maxverro - Row 2 and 3 (DecimalExpression and IntegerExpression) were fixed in 5.8.8, are you using an older version?

The other two (_HRForException and Operator::ToString()) have now been removed from the codebase in our develop branch (see the commit below) and PEVerify now validates EPPlus.dll without errors.

![image](https://user-images.githubusercontent.com/897655/187875961-593ffdcd-fb53-48d2-84d9-e17948000c39.png)

I will fix this in our branch for EPPlus 5 too, so it will be included in our next revision (6.0.8/5.8.14).

**Comment by swmal** [comment]
The failed build indicator on the last commit can be ignored. Here is a link to the latest build for EPPlus 5: https://ci.appveyor.com/project/EPPlusSoftware/epplus5

This will be released in the next version for EPPlus 5 and 6.

**Comment by swmal** [comment]
The adjustments we have made to get rid of these warnings are included in 5.8.14 and 6.0.8 (released on Nuget today).
